### PR TITLE
[stdlib] Comparison benchmarks for String

### PR DESCRIPTION
**Pull request by lancep**
Adds a few benchmarks for string comparisons. This is in preparation for the revamped string comparison and will give us a look at the before/after

### CODE REVIEW

**Comment by lancep** [comment]
@swift-ci please benchmark

**Comment by lancep** [comment]
@swift-ci please smoke test and merge

File: benchmark/single-source/StringComparison.swift.gyb [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+  public func run_StringComparison_${Name}(_ N: Int) {
+    var count = 0
+    let workload = Workload.${Name}
+    let tripCount = workload.tripCount
+    let payload = workload.payload
+    for _ in 1...tripCount*N {
+      for s1 in payload {
+        for s2 in payload {
+          let cmp = s1 < s2
+          count += cmp ? 1 : 0
```
Comment (milseman) [inline benchmark/single-source/StringComparison.swift.gyb]: These days, the benchmark suite has a `blackHole` function (inline never in another module) that you can also feed results into. That's an additional guarantee that the entire benchmark won't be trivialized by a better optimizer. (This is why many benchmarks followed the pattern of an inline(never) helper that returned a result derived from the inner loop, though that wasn't always perfect and `blackHole` is better).

**Review by milseman — COMMENTED** [review]
👍 

Could you also have a workload or two that has shared long, common prefixes? E.g. comparison between two long URLs on the same domain, etc.

**Pull request merged**